stestr/subunit_trace.py

```diff
@@ -470,24 +466,29 @@ def trace(
             for x in RESULTS[worker]
             if x["timestamps"][1] is not None
         ]
-    if not start_times:
+    if print_full_output(stdout, start_times, stop_times, post_fails, no_summary):
+        return 1
+    return 0 if results.wasSuccessful(summary) else 1
+
+
+def print_full_output(stdout, start_times, stop_times, post_fails, no_summary):
+    """Print output plus edge case validation"""
+    if not start_times or count_tests("status", ".*") == 0:
         print("The test run didn't actually run any tests", file=sys.stderr)
         return 1
+
     start_time = min(start_times)
     stop_time = max(stop_times)
     elapsed_time = stop_time - start_time
 
-    if count_tests("status", ".*") == 0:
-        print("The test run didn't actually run any tests", file=sys.stderr)
-        return 1
     if post_fails:
         print_fails(stdout)
     if not no_summary:
         print_summary(stdout, elapsed_time)
 
     # NOTE(mtreinish): Ideally this should live in testtools streamSummary
     # this is just in place until the behavior lands there (if it ever does)
-    if count_tests("status", "^success$") == 0:
+    if count_tests("status", "^xfail$") + count_tests("status", "^success$") == 0:
```

Comment:
figured this is easier to read than join the two regex

Reply:
yeah, that's fine I think the regex would just be `^xfail|success$` or something like that, but this is definitely more legible.